model/base/src/main/java/org/eclipse/ditto/model/base/headers/AbstractDittoHeaders.java

```diff
@@ -148,23 +146,11 @@ public JsonObject toJson() {
     }
 
     private Class<?> getTypeForKey(final CharSequence key) {
-        final Collection<HeaderDefinition> definitions = getDefinitions();
-
-        return definitions.parallelStream()
-                .filter(definition -> Objects.equals(definition.getKey(), key))
-                .map(HeaderDefinition::getJavaType)
-                .findAny()
+        return DittoHeaderDefinition.forKey(key)
```

Comment:
Does this solution still work with for example `ImmutableMessageHeaders` which have additional HeaderDefinitions to `DittoHeaderDefinition`? Line 163 of the replaced code did take this into account.

Reply:
Indeed, all values of headers not defined in `DittoHeaderDefinitions` are serialized as JSON strings. However, the code behaves that way also before this pull request. I will make a separate issue for it.

Reply:
Oops, actually `MessageHeaders` uses `getSpecificDefinitions()` to serialize extra non-string headers. Thanks for pointing it out; I will add a fix.